File: problem_sets/pset6/dont-panic-python/hack.py
from cs50 import SQL

# Establish the connection to the dont-panic database.
db = SQL("sqlite:///dont-panic.db")

# Create a variable to store the user defined password.
password = input("Please, enter a password: ")

# Execute sql statement with Python, ensuring to use the
# '?', from the cs50's library, as placeholder (prepared statement),
# for a value to supply later.
db.execute(
    """
    UPDATE "users"
    SET "password" = ?
    WHERE "username" = 'admin';
    """,
    # Tell the execute statement which value to replace
    # the question mark with.
    password
)

File: problem_sets/pset6/dont-panic-java/Hack.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.util.Scanner;

public class Hack {
    public static void main(String[] args) throws Exception {
        // Store the user's input (password) inside a variable 'password'.
        Scanner scanner = new Scanner(System.in);
        System.out.println("Please, enter the new password: ");
        String password = scanner.nextLine();
        
        // Create a connection between the database and the program.
        Connection sqliteConnection = DriverManager.getConnection("jdbc:sqlite:dont-panic.db");
    
        // Store the sql query inside a 'java' string.
        String query = """
            UPDATE "users"
            SET "password" = ?
            WHERE "username" = 'admin';
        """;

        // Execute sql statement using 'java', ensuring to use the
        // '?', as placeholder (prepared statement), for a value to supply later.
        PreparedStatement sqliteStatement  = sqliteConnection.prepareStatement(query);
        sqliteStatement.setString(1, password); // Insert the password.
        sqliteStatement.executeUpdate();

        // Close connection
        sqliteConnection.close();
        
        scanner.close();
    }
}
